src/core/meshcop/tcat_agent.cpp

```diff
@@ -535,6 +547,86 @@ Error TcatAgent::HandlePing(const Message &aIncomingMessage,
     }
 
     SuccessOrExit(error = aOutgoingMessage.AppendBytesFromMessage(aIncomingMessage, aOffset, aLength));
+    response = true;
+
+exit:
+    return error;
+}
+
+Error TcatAgent::HandleGetNetworkName(Message &aOutgoingMessage, bool &response)
+{
+    Error             error    = kErrorNone;
+    MeshCoP::NameData nameData = Get<MeshCoP::NetworkNameManager>().GetNetworkName().GetAsData();
+
+    VerifyOrExit(Get<ActiveDatasetManager>().IsCommissioned(), error = kErrorInvalidState);
+#if !OPENTHREAD_CONFIG_ALLOW_EMPTY_NETWORK_NAME
+    VerifyOrExit(nameData.GetLength() > 0, error = kErrorInvalidState);
+#endif
+
+    SuccessOrExit(
+        error = Tlv::AppendTlv(aOutgoingMessage, kTlvResponseWithPayload, nameData.GetBuffer(), nameData.GetLength()));
+    response = true;
+
+exit:
+    return error;
+}
+
+Error TcatAgent::HandleGetDeviceId(Message &aOutgoingMessage, bool &response)
+{
+    uint8_t        *deviceId;
+    uint16_t        length = 0;
+    Mac::ExtAddress eui64;
+    Error           error = kErrorNone;
+
+    if (mVendorInfo->mGeneralDeviceId != nullptr)
+    {
+        length   = mVendorInfo->mGeneralDeviceId->mDeviceIdLen;
+        deviceId = const_cast<uint8_t *>(mVendorInfo->mGeneralDeviceId->mDeviceId);
```

Comment:
I think we can declare `deviceId` as `const uint8_t *`. We are not modifying the buffer pointed to by `deviceId` in the code (which `const` would prevent), but we can still change the value of the `deviceId` pointer itself.
* This eliminates the need to cast away `const`-ness here.
* The assignment `deviceId = eui64.m8` should still work as expected.

Reply:
fixed.